### PR TITLE
Add `size` to Scatter

### LINKED ISSUE #13957: Explicit selection glyph example broken
ref: https://docs.bokeh.org/en/latest/docs/user_guide/styling/plots.html#selected-and-unselected-glyphs

Code there does this:
```python
plot = figure(width=400, height=400, tools="tap", title="Select a circle")
renderer = plot.scatter([1, 2, 3, 4, 5], [2, 5, 8, 2, 7], size=50)

selected_scatter = Scatter(fill_alpha=1, fill_color="firebrick", line_color=None)
nonselected_scatter = Scatter(fill_alpha=0.2, fill_color="blue", line_color="firebrick")

renderer.selection_glyph = selected_scatter
renderer.nonselection_glyph = nonselected_scatter
```

Original plot:

Screenshot 2024-07-01 at 09 01 45

After tap (size is wrong):

Screenshot 2024-07-01 at 09 01 58

Previous version result:

Screenshot 2024-07-01 at 09 02 26

cc @mattpap I expect this is related to glyph inheritance changes?

### PR DESCRIPTION
**Pull request by mosc9575**
This PR fixes the [explicit selection glyph example](https://docs.bokeh.org/en/latest/docs/examples/styling/plots/glyph_selection_models.html) by adding size parameter to the Scatter model, which is by default 4.

- [ ] issues: fixes #13957

Here the result with the fix using bokeh 3.4.2 in a jupyter notebook after a click.

grafik

### CODE REVIEW

**Comment by bryevdv** [comment]
I think the best solution here is for the scatter sub-glyphs to inherit size by default if the user does not override it explicitly. I.e. it should be possible but not required to override the main glyph value, *for sub-glyphs*. That's how other properties work for sub-glyphs.

**Comment by mosc9575** [comment]
I tried to solve the problem and updated the javascript code to inherit from the base glyph.

For the example the visual output meets my expectations.

```python
from bokeh.models import Scatter, Circle, ColumnDataSource
from bokeh.resources import Resources
from bokeh.plotting import figure, show, output_notebook

source = ColumnDataSource(dict(xx=[1, 2, 3, 4, 5],yy=[2, 5, 8, 2, 7]))

plot = figure(width=400, height=400, tools="tap", title="Select a circle")
renderer = plot.scatter('xx', 'yy', size=50, source=source)

selected_scatter = Circle(fill_alpha=1, fill_color="firebrick", radius=0.1)
nonselected_scatter = Scatter(fill_alpha=0.2, fill_color="blue", line_color="firebrick")

renderer.selection_glyph = selected_scatter
renderer.nonselection_glyph = nonselected_scatter

show(plot)
```

select_a_circle

The problem is, that the models on the Python side are not updated and I don't know how I can solve this.

@bryevdv and @mattpap Help is needed. Any suggestions on the code and how to solve this for python are welcome. I will try finish this PR afterwards.

**Comment by bryevdv** [comment]
@mosc9575 thanks for your patience, please take a look at https://github.com/bokeh/bokeh/issues/13957#issuecomment-2308634367 for what I think is a better short-term "python only" solution that does not burden the user with manually keeping properties in sync, and does not burden us with technical lifts. 

To fully implement this, we'll want to search and update all examples that set sub-glyphs using the `bokeh.models` API to follow that pattern with `clone`. Please let me know whether you have the bandwidth continue with these updates here. I understand, if not, but then I will close this PR.

**Comment by mosc9575** [comment]
@bryevdv I will try to do this task in the next few days.

File: examples/advanced/extensions/parallel_plot/parallel_plot.py [hunk 1]
```diff
@@ -1,9 +1,8 @@
 import numpy as np
 
 from bokeh.layouts import column
-from bokeh.models import (BasicTickFormatter, ColumnDataSource,
-                          CustomJSTickFormatter, Div, FixedTicker,
-                          LinearAxis, LinearColorMapper, MultiLine, Range1d)
+from bokeh.models import (BasicTickFormatter, ColumnDataSource, CustomJSTickFormatter,
+                          Div, FixedTicker, LinearAxis, LinearColorMapper, Range1d)
```
Comment (mosc9575) [inline examples/advanced/extensions/parallel_plot/parallel_plot.py:5]: While working on some examples I also fixed some PEP-Warnings and had to rearange some lines for the code style checks. This is why some files have "many" changes.

File: examples/interaction/js_callbacks/customjs_for_tools.py [hunk 2]
```diff
@@ -21,10 +21,15 @@
...
-quad = Quad(left='left', right='right',top='top', bottom='bottom',
+quad = Quad(left='left', right='right', top='top', bottom='bottom',
             fill_alpha=0.3, fill_color='#009933')
 
-p.add_glyph(source, quad, selection_glyph=quad, nonselection_glyph=quad)
+p.add_glyph(
+    source,
+    quad,
+    selection_glyph=quad.clone(fill_color='blue'),
+    nonselection_glyph=quad.clone(fill_color='gray'),
```
Comment (mosc9575) [inline examples/interaction/js_callbacks/customjs_for_tools.py:31]: With this change and adding the "TapTool" to the tools, this example makes more sense to me. In the newer version a click on a renderer has a visual effect.

File: src/bokeh/plotting/_figure.py [hunk 3]
```diff
@@ -369,7 +369,7 @@ def harea_stack(self, stackers, **kw):
         Examples:
 
             Assuming a ``ColumnDataSource`` named ``source`` with columns
-            *2016* and *2017*, then the following call to ``harea_stack`` will
```
Comment (mosc9575) [inline src/bokeh/plotting/_figure.py:372]: The consecutive "will" is fixed, but this fix is unrelated to the original issue.

**Review by bryevdv — APPROVED** [review]

File: examples/advanced/extensions/parallel_plot/parallel_plot.py (hunk 1 above)
Comment (mattpap) [reply examples/advanced/extensions/parallel_plot/parallel_plot.py:5]: It would nice if we could enable the same style of imports, that we use under `src/`, for when we need to import a lot of models, etc.

**Review by mattpap — APPROVED** [review]

File: examples/advanced/extensions/parallel_plot/parallel_plot.py (hunk 1 above)
Comment (mosc9575) [reply examples/advanced/extensions/parallel_plot/parallel_plot.py:5]: If I change the style of the imports the pre-push-hook `test_isort` fails.

We could add [`-m=3`](https://pycqa.github.io/isort/docs/configuration/multi_line_output_modes.html) to the line below

https://github.com/bokeh/bokeh/blob/df57080f22b39816deee84d5e835f752d1713925/tests/codebase/test_isort.py#L58

but this woul mean, we have to change all import in all files. This would be worth a new issue. Is it okey for you if I open a seperate PR, to make this change to all import in this project?

Comment (bryevdv) [reply examples/advanced/extensions/parallel_plot/parallel_plot.py:5]: The import style under `examples` was made different from that under `src` on purpose. We can discuss whether that should be the case or not any longer, but we should not take any action regarding that in this PR. In any was there should be an issue to discuss and decide the course of action first.
